Bootloader/bootloader_core.c

```diff
@@ -0,0 +1,514 @@
+}
+#endif
+
+#ifdef BOOTLOADER_CONFIG
+/******************************************************************************
+ * @brief set state of the bootloader
+ * @param state 
+ * @return None
+ ******************************************************************************/
+void LuosBootloader_SetState(bootloader_state_t state)
+{
+    bootloader_state = state;
+}
+#endif
+
+#ifdef BOOTLOADER_CONFIG
+/******************************************************************************
+ * @brief Bootloader machine state
+ * @param None
+ * @return None
+ ******************************************************************************/
+void LuosBootloader_Task(void)
+{
+    switch (bootloader_state)
+    {
+        case BOOTLOADER_START_STATE:
+            // set ID node saved in flash
+            LuosBootloader_SetNodeID();
+            // go to ready state
+            LuosBootloader_SetState(BOOTLOADER_READY_STATE);
+            break;
+
+        case BOOTLOADER_READY_STATE:
+            // if STOP_CMD, restart the node
+            if (bootloader_cmd == BOOTLOADER_STOP)
+            {
+                LuosBootloader_SetState(BOOTLOADER_STOP_STATE);
+            }
+            // if READY_CMD, continue BOOTLOADER process
+            if (bootloader_cmd == BOOTLOADER_READY)
+            {
+                // save binary length
+                memcpy(&nb_bytes, &bootloader_data, sizeof(uint32_t));
+                // check free space in flash
+                if (LuosBootloader_IsEnoughSpace(nb_bytes))
+                {
+                    // send READY response
+                    LuosBootloader_SendResponse(BOOTLOADER_READY_RESP);
+                    // go to HEADER state
+                    LuosBootloader_SetState(BOOTLOADER_ERASE_STATE);
+                }
+                else
+                {
+                    // send READY response
+                    LuosBootloader_SendResponse(BOOTLOADER_ERROR_SIZE);
+                }
+            }
+            break;
+
+        case BOOTLOADER_ERASE_STATE:
+
+            if (bootloader_cmd == BOOTLOADER_ERASE)
+            {
+                // erase flash memory
+                LuosBootloader_EraseMemory();
+                // send READY response
+                LuosBootloader_SendResponse(BOOTLOADER_ERASE_RESP);
+                // go to HEADER state
+                LuosBootloader_SetState(BOOTLOADER_BIN_CHUNK_STATE);
+            }
+
+            break;
+
+        case BOOTLOADER_BIN_CHUNK_STATE:
+
+            if (bootloader_cmd == BOOTLOADER_BIN_CHUNK)
+            {
+                // reset bootloader cmd to avoid looping in this portion of the code
+                bootloader_cmd = BOOTLOADER_IDLE;
+                // handle binary data
+                LuosBootloader_ProcessData();
+                // send ack to the Host
+                LuosBootloader_SendResponse(BOOTLOADER_BIN_CHUNK_RESP);
+            }
+
+            if (bootloader_cmd == BOOTLOADER_BIN_END)
```

Comment:
by using Luos_ReceiveData you don't need it anymore.

Reply:
Ok I get it. It's because Luos_ReceiveData can't manage more than 64Kb data.

Reply:
I don't get it, we don't receive data here, we just save in flash the last chunk of data received the previous message.